bids_conversion/convert_somato_data.py

```diff
@@ -0,0 +1,258 @@
+
+# Get the events, and assign an ID
+events = mne.find_events(raw)
+event_id = dict()
+for val in np.unique(events[:, -1]):
+    event_id['somato_event{}'.format(val)] = val
+
+print('\n\n')
+print('event_id: "{}"'.format(event_id))
+
+# Make a bids name for the raw file
+bids_basename = make_bids_basename(subject='01', task='somato')
+
+# Make a location for the BIDS formatted data
+somato_parent, somato_name = op.split(somato_path)
+somato_path_bids = op.join(somato_parent, somato_name + '-bids')
+
+# Write the data
+write_raw_bids(raw, bids_basename, somato_path_bids, events, event_id,
+               overwrite=True)
+
+# Edit dataset_description.json to include link to dataset
+dataset_description_json = op.join(somato_path_bids,
+                                   'dataset_description.json')
+with open(dataset_description_json, 'r') as fin:
+    jsondict = json.load(fin)
+
+# Edit some fields we know of
+jsondict['Name'] = 'MNE-somato-data-bids'
+jsondict['Authors'] = ['Lauri Parkkonen']
+jsondict['License'] = 'PDDL, see: https://opendatacommons.org/licenses/pddl/'
+jsondict['ReferencesAndLinks'] = ['http://martinos.org/mne/stable/manual/datasets_index.html?#somatosensory']  # noqa: E501
+jsondict['Acknowledgements'] = "Stefan Appelhoff, Alexandre Gramfort, and Mainak Jas formatted these data to BIDS."   # noqa: E501
+
+
+# overwrite the json, updating it
+with open(dataset_description_json, 'w') as fout:
+    json.dump(jsondict, fout, indent=2, sort_keys=True)
+
+# Write anatomical data to BIDS
+# we take the original T1 from the freesurfer directory
+t1w = op.join(somato_path, 'subjects', 'somato', 'mri', 'T1.mgz')
+
+# we also take the trans file, and use it to write the coordinates of
+# anatomical landmarks to a T1w.json file
+trans = op.join(somato_path, 'MEG', 'somato', 'sef_raw_sss-trans.fif')
+
+# Copy over the MRI, convert it to NIfTI format, and write the anatomical
+# landmarks in voxel coordinates
+anat_dir = write_anat(somato_path_bids, subject='01', t1w=t1w, trans=trans,
+                      raw=raw, overwrite=True, verbose=True)
+t1w_nii = op.join(anat_dir, 'sub-01_T1w.nii.gz')
+
+# Add derivatives
+# some files cannot be specified in BIDS yet. We add these to derivatives
+# General derivatives directory
+derivatives_dir = op.join(somato_path_bids, 'derivatives')
+if not op.exists(derivatives_dir):
+    os.makedirs(derivatives_dir)
+
+# Make a freesurfer directory and copy over derivatives: Consider the
+# instructions in the docstring at the top of the module
+subjects_dir_bids = op.join(derivatives_dir, 'freesurfer', 'subjects')
+if not op.exists(subjects_dir_bids):
+    os.makedirs(subjects_dir_bids)
+
+# Run recon-all from freesurfer
+run_subprocess(['recon-all', '-i', t1w_nii, '-s', '01', '-all'])
+
+# Run make_scalp_surfaces
+run_subprocess(['mne', 'make_scalp_surfaces', '-s', '01', '--overwrite'])
+
+# Run watershed_bem
+run_subprocess(['mne', 'watershed_bem', '-s', '01', '--overwrite'])
```

Comment:
summoning @larsoner @jasmainak @agramfort to discuss the lines above (155 to 162):

Could you explain to me which freesurfer commands (and/or their mne-bindings like `mne watershed_bem`) apart from `recon-all` I need to run to reproduce what we have in `MNE-somato-data/subjects/somato`?

I guessed now from the `MNE-somato-data/subjects/somato/scripts` directory and the discussions we had on Gitter:

- I definitely need `recon-all`
- there is a `mkheadsurf.log` in /subjects/somato/scripts ... so I probably need `mne make_head_surfaces` after `recon-all`
- there is a `bem` directory in /subjects/somato ... so I probably need `mne watershed_bem` after `recon-all`

1. is that true?
1. do I need scripts beyond these to reproduce the sample data?
1. does it matter whether I run `watershed_bem` or `make_scalp_surfaces` first?


The MNE docs don't seem to help me a lot on this. I have looked at:

- https://martinos.org/mne/stable/auto_tutorials/source-modeling/plot_background_freesurfer.html#freesurfer-integration-with-mne-python
- https://surfer.nmr.mgh.harvard.edu/fswiki/RecommendedReconstruction (this just mentions `recon-all`)
- https://martinos.org/mne/stable/manual/cookbook.html

Reply:
Let me share with you privately on email a document I have from Lauri. You may find it useful

Reply:
okay answering your questions as well:
>  is that true?

seems correct

> do I need scripts beyond these to reproduce the sample data?

I don't think so. But you're unlikely to reproduce the exact same thing as Lauri might have used different software

> does it matter whether I run watershed_bem or make_scalp_surfaces first?

I don't think so